### PR TITLE
Integrating Realm in an Android App

### PR DESCRIPTION
**Pull request by echessa**

### CODE REVIEW

**Comment by echessa** [comment]
Hi @kukicado I wanted to check in on this.

File: _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown [hunk 1]
```diff
@@ -0,0 +1,1080 @@
...
+tags: 
+- android
+- realm
+related:
+- 2015-06-29-using-smartlock-on-android-in-3-simple-steps
+- 2016-02-08-how-to-authenticate-on-android-using-social-logins
+---
+
+## Introduction
+The [Realm Mobile Database](https://realm.io/docs/get-started/overview/#the-realm-mobile-database) is a cross-platform database solution that can be used as an alternative to SQLite and Core Data. Compared to these two options, Realm is easier to set up and use. To perform the same operation in Realm, you usually end up writing fewer lines of code than you would with SQLite or Core Data. On performance, Realm is said to be faster and it also offers other modern features such as encryption, JSON support and data change notifications.
```
Comment (kukicado) [inline _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown]: Link to SQLite and Core Data homepages.

File: _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown [hunk 2]
```diff
@@ -0,0 +1,1080 @@
...
+
+Unlike a traditional database, objects in Realm are native objects. You don’t have to copy objects out of the database, modify them, and save them back—you’re always working with the “live,” real object. If one thread or process modifies an object, other threads and processes can be immediately notified. Objects always stay in sync.
+
+If your application needs to store user data to the cloud and have it synced on all devices used by the user, you can use the Realm Mobile Database together with the [Realm Object Server](https://realm.io/docs/realm-object-server/) for this. In this article though, we are going to focus on the Realm Mobile Database. We are going to see how to integrate it into an Android app and perform the usual CRUD operations on it. We'll create a To Do application which will enable the user to create, edit and delete tasks from a list.
```
Comment (kukicado) [inline _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown]: IS there a link to Realm Mobile Database?

File: _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown [hunk 3]
```diff
@@ -0,0 +1,1080 @@
...
+
+## Getting Started
+To get started, first create an Android project (I named mine Tasky). Select a Basic Activity template for it and on the last window of the project creation wizard, change the **Activity Name** to `TaskListActivity`.
```
Comment (kukicado) [inline _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown]: Can you mention which software you're using? Android Studio?

File: _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown [hunk 4]
```diff
@@ -0,0 +1,1080 @@
...
+
+String (`String`) and boxed integer (`Byte`, `Short`, `Integer`, and `Long`) Primary keys can have `null` values and so we also mark `id` with `@Required` to enforce a check.
+
+## Working with RealmBaseAdapter
+
+Our application will display the Task items in a ListView, and to work with this, we need an adapter that will manage the data model and adapt it to individual rows in the ListView.
+
+Realm makes available two adapters that can be used to bind its data to UI widgets, in particular data coming from `OrderedRealmCollection` (`RealmResults` and `RealmList`, which we'll look at shortly, implement this interface). There is the `RealmBaseAdapter` for working with ListViews and `RealmRecyclerViewAdapter` for working with RecyclerViews.
```
Comment (kukicado) [inline _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown]: available two adapters => two adapters available

File: _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown (hunk 1 above)
Comment (echessa) [inline _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown]: Fixed

File: _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown (hunk 2 above)
Comment (echessa) [inline _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown]: Added link

File: _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown (hunk 3 above)
Comment (echessa) [inline _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown]: Done

File: _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown (hunk 4 above)
Comment (echessa) [inline _posts/2017-03-13-integrating-realm-database-in-an-android-application.markdown]: Fixed

**Pull request merged**